test/test.js

```diff
@@ -207,24 +207,36 @@ describe('Filter', function() {
 
       expect(existsSync(fileForRemoval)).to.be.false;
       expect(existsSync(results.directory + '/a/README.md')).to.be.true;
+      expect(existsSync(results.subject.cachePath + '/a/README.md'), 'OUTPUT: a/README.md should NO LONGER be present').to.be.true;
 
       return results.builder();
     }).then(function(results) {
-      expect(existsSync(results.directory + '/a/README.md'), 'OUTPUT: a/foo.js should NO LONGER be present').to.be.false;
 
-      expect(existsSync(fileForRemoval)).to.be.false;
+     //  expect(existsSync(results.directory + '/a/README.md'), 'OUTPUT: a/README.md should NO LONGER be present').to.be.false;
+       expect(existsSync(results.subject.cachePath + '/a/README.md'), 'OUTPUT: a/README.md should NO LONGER be present').to.be.false;
```

Comment:
From my understanding, This test tests if the file is deleted from cache (should purge cache). So we should check whether the file exists in cache after it is deleted and if it is present again when something is written in the file?